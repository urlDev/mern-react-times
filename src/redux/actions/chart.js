import axios from "axios";

export const FETCH_FOREX_SUCCESS = "FETCH_FOREX_SUCCESS";
export const FETCH_CHART_ERROR = "FETCH_CHART_ERROR";
export const SET_LOADING = "SET_LOADING";
export const SET_LOADING_TRUE = "SET_LOADING_TRUE";
export const CHANGE_MARKET_TYPE = "CHANGE_MARKET_TYPE";
export const CHANGE_MARKET_NAME = "CHANGE_MARKET_NAME";
export const SET_MARKET_DETAIL = "SET_MARKET_DETAIL";
export const SET_CHART_DATA = "SET_CHART_DATA";
export const CLEAN_CHART_DATA = "CLEAN_CHART_DATA";
export const SET_CHART_TIME_FRAME = "SET_CHART_TIME_FRAME";
export const SET_RATING = "SET_RATING";
export const SET_HOME_CHART_DATA = "SET_HOME_CHART_DATA";
export const CLEAN_STATE = "CLEAN_STATE";
export const GET_SEARCH_RESULTS = "GET_SEARCH_RESULTS";
export const SET_SEARCH_MARKET_DETAIL = "SET_SEARCH_MARKET_DETAIL";
export const CLEAR_SEARCH_RESULTS = "CLEAR_SEARCH_RESULTS";
export const OPEN_SEARCH_MODAL = "OPEN_SEARCH_MODAL";
export const CLOSE_SEARCH_MODAL = "CLOSE_SEARCH_NODAL";

export const closeSearchModal = () => ({
  type: CLOSE_SEARCH_MODAL,
});

export const openSearchModal = () => ({
  type: OPEN_SEARCH_MODAL,
});

export const clearSearchResults = () => ({
  type: CLEAR_SEARCH_RESULTS,
});

export const setSearchMarketDetail = (details) => ({
  type: SET_SEARCH_MARKET_DETAIL,
  payload: details,
});

export const getSearchResults = (result) => ({
  type: GET_SEARCH_RESULTS,
  payload: result,
});

export const setLoading = () => ({
  type: SET_LOADING,
});

export const setLoadingTrue = () => ({
  type: SET_LOADING_TRUE,
});

export const cleanState = () => ({
  type: CLEAN_STATE,
});

export const setHomeChartData = (chart) => ({
  type: SET_HOME_CHART_DATA,
  payload: [chart],
});

export const setRating = (rating) => ({
  type: SET_RATING,
  payload: rating,
});

export const setChartTimeFrame = (time) => ({
  type: SET_CHART_TIME_FRAME,
  payload: time,
});

export const setChartData = (data) => ({
  type: SET_CHART_DATA,
  payload: data,
});

export const cleanChartData = () => ({
  type: CLEAN_CHART_DATA,
});

export const setMarketDetail = (market) => ({
  type: SET_MARKET_DETAIL,
  payload: market,
});

export const changeMarketType = (market) => ({
  type: CHANGE_MARKET_TYPE,
  payload: market,
});

export const changeMarketName = (market) => ({
  type: CHANGE_MARKET_NAME,
  payload: market,
});

export const fetchForexSuccess = (chart) => ({
  type: FETCH_FOREX_SUCCESS,
  payload: chart,
});

export const fetchChartError = (error) => ({
  type: FETCH_CHART_ERROR,
  payload: error,
});

export const fetchForex = (market) => async (dispatch) => {
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/quote/${market}?apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = await response.data;
    // FinancialModel API doesn't send error if it's about limit reach.
    // In that sense, they send an error message with 200 status. Weird.
    // Thats why I am catching that error here and dispatching it.
    if (data.length) {
      return data.length === 1
        ? [dispatch(setSearchMarketDetail(data))]
        : dispatch(fetchForexSuccess(data));
    } else {
      return dispatch(fetchChartError(data));
    }
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

export const fetchChartData = (symbol, timeFrame) => async (dispatch) => {
  dispatch(cleanChartData());
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/historical-chart/${timeFrame}/${symbol}?apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = await response.data;
    return dispatch(setChartData(data));
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

export const fetchRating = (symbol) => async (dispatch) => {
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/rating/${symbol}?apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = await response.data;
    return dispatch(setRating(data));
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

/*
This deserves to be mentioned.
Few months ago, when I was making the first version of this app, I used array.map to fetch charts.
Because of the async nation and using array map method, response was coming irregular and I had to sort them.
Now, with using for of loop, I dont need to write any extra code because loop waits for one to finish to move on.
Plus, its so clean!
Yet another homerun for me! 😁😎
*/
export const fetchHomeChart = (symbols) => async (dispatch) => {
  const symbolsArray = symbols.split(",").sort();
  let charts = [];
  try {
    // dispatch(setLoadingTrue());
    dispatch(cleanState());
    for (let symbol of symbolsArray) {
      const response = await axios.get(
        `https://financialmodelingprep.com/api/v3/historical-chart/1hour/${symbol}?apikey=${process.env.REACT_APP_CHART_KEY}`
      );
      const data = response.data;
      charts.push(data);
    }
    return charts.length
      ? dispatch(setHomeChartData(charts))
      : dispatch(fetchChartError(charts));
    // dispatch(setLoading());
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

export const fetchSearch = (input) => async (dispatch) => {
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/search?query=${input}&limit=10&apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = response.data;
    return dispatch(getSearchResults(data));
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};
