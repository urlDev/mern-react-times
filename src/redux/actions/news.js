import axios from 'axios';

export const FETCH_POPULAR_SUCCESS = 'FETCH_POPULAR_SUCCESS';
export const FETCH_STORY_SUCCESS = 'FETCH_STORY_SUCCESS';
export const FETCH_NEWS_ERROR = 'FETCH_NEWS_ERROR';
export const CHANGE_HEADER = 'CHANGE_HEADER';

export const fetchPopularSuccess = (news) => ({
    type: FETCH_POPULAR_SUCCESS,
    payload: news,
});

export const fetchStorySuccess = (news) => ({
    type: FETCH_STORY_SUCCESS,
    payload: news,
});

export const changeHeader = (header) => ({
    type: CHANGE_HEADER,
    payload: header,
});

export const fetchNewsError = (error) => ({
    type: FETCH_NEWS_ERROR,
    payload: error,
});

export const fetchTopStories = (topStories) => async(dispatch) => {
    try {
        const response = await axios.get(
            `https://api.nytimes.com/svc/topstories/v2/${topStories}.json?api-key=${process.env.REACT_APP_NEWS_API_KEY}`,
        );
        const data = await response.data;
        return dispatch(fetchStorySuccess(data));
    } catch (error) {
        return dispatch(fetchNewsError(error));
    }
};

export const fetchMostPopular = () => async(dispatch) => {
    try {
        const response = await axios.get(
            `https://api.nytimes.com/svc/mostpopular/v2/viewed/7.json?api-key=${process.env.REACT_APP_NEWS_API_KEY}`,
        );
        const data = await response.data;
        return dispatch(fetchPopularSuccess(data));
    } catch (error) {
        return dispatch(fetchNewsError(error));
    }
};