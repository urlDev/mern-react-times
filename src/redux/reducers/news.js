import {
  FETCH_NEWS_ERROR,
  FETCH_POPULAR_SUCCESS,
  FETCH_STORY_SUCCESS,
  CHANGE_HEADER,
  CLEAN_STATE,
  CLEAR_ERROR,
  SET_WIDTH,
  OPEN_RESPONSIVE_MENU,
  CLOSE_RESPONSIVE_MENU,
} from "../actions/news";

const initialState = {
  loadingNews: true,
  popular: [],
  story: [],
  errorNews: null,
  header: "home",
  width: window.innerWidth,
  responsiveMenu: false,
};

const newsReducer = (state = initialState, action) => {
  switch (action.type) {
    case CLEAR_ERROR:
      return {
        ...state,
        errorNews: null,
      };
    case FETCH_POPULAR_SUCCESS:
      return {
        ...state,
        loadingNews: false,
        popular: action.payload.results.slice(0, 4),
      };
    case FETCH_STORY_SUCCESS:
      return {
        ...state,
        loadingNews: false,
        story: action.payload.results.slice(0, 3),
      };
    case FETCH_NEWS_ERROR:
      return {
        ...state,
        errorNews: action.payload,
      };
    case CHANGE_HEADER:
      return {
        ...state,
        header: action.payload,
      };
    case SET_WIDTH:
      return {
        ...state,
        width: action.payload,
      };
    case OPEN_RESPONSIVE_MENU:
      return {
        ...state,
        responsiveMenu: true,
      };
    case CLOSE_RESPONSIVE_MENU:
      return {
        ...state,
        responsiveMenu: false,
      };
    default:
      return state;
  }
};

export default newsReducer;
