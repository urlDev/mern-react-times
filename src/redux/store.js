import { createStore, applyMiddleware } from 'redux';
import rootReducer from './rootReducer';

import thunk from 'redux-thunk';
import logger from 'redux-logger';

const middlewares = [thunk, logger];

export default createStore(rootReducer, applyMiddleware(...middlewares));