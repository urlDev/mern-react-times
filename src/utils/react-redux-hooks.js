// Mocking redux hooks
import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from "react-redux";

export const useSelector = (state) => originalUseSelector(state);
export const useDispatch = () => originalUseDispatch();
