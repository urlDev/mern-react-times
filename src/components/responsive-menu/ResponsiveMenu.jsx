import { useDispatch, useSelector } from "utils/react-redux-hooks";

import { openResponsiveMenu } from "redux/actions/news";

import HamburgerSrc from "assets/hamburger.svg";
import HamburgerCloseSrc from "assets/hamburgerClose.svg";

import { HamburgerMenu } from "./ResponsiveMenu.styles";

const ResponsiveMenu = ({ icon }) => {
  const { responsiveMenu } = useSelector((news) => news.news);
  const dispatch = useDispatch();

  return (
    <HamburgerMenu
      icon={icon}
      alt="hamburger menu"
      onClick={() => dispatch(openResponsiveMenu())}
    >
      <img
        src={responsiveMenu ? HamburgerCloseSrc : HamburgerSrc}
        alt="hamburger icon"
      />
    </HamburgerMenu>
  );
};

export default ResponsiveMenu;
