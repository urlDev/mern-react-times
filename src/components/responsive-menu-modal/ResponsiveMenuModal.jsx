import { useDispatch } from "../../utils/react-redux-hooks";

import {
  changeHeader,
  clearError,
  closeResponsiveMenu,
} from "../../redux/actions/news";
import { topics } from "../story-topic-headers/StoryTopicHeaders";

import {
  MenuModalAllPage,
  MenuModalContainer,
} from "./ResponsiveMenuModal.styles";

import { StyledLink } from "../story-topic-headers/StoryTopicHeaders.styles";

const ResponsiveMenuModal = () => {
  const dispatch = useDispatch();

  return (
    <MenuModalAllPage onClick={() => dispatch(closeResponsiveMenu())}>
      <MenuModalContainer>
        {topics.map((topic) => (
          <StyledLink key={topic} to={`/${topic.toLowerCase()}`}>
            <span
              onClick={() => {
                dispatch(changeHeader(topic.toLowerCase()));
                dispatch(closeResponsiveMenu());
                dispatch(clearError());
              }}
            >
              {topic}
            </span>
          </StyledLink>
        ))}
      </MenuModalContainer>
    </MenuModalAllPage>
  );
};

export default ResponsiveMenuModal;
