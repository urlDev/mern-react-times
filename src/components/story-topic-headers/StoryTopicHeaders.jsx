import moment from "moment";
import { useDispatch, useSelector } from "react-redux";

import { changeHeader, clearError } from "../../redux/actions/news";

import ResponsiveMenu from "../responsive-menu/ResponsiveMenu";
import ResponsiveMenuModal from "../responsive-menu-modal/ResponsiveMenuModal";

import {
  StoryTopicHeadersContainer,
  TopicContainer,
  Date,
  StyledLink,
} from "./StoryTopicHeaders.styles";

export const topics = [
  "Home",
  "Arts",
  "Business",
  "Health",
  "Opinion",
  "Politics",
  "Science",
  "Sports",
  "Technology",
  "Travel",
];

const StoryTopics = () => {
  const { width, responsiveMenu } = useSelector((news) => news.news);
  const dispatch = useDispatch();

  return (
    <StoryTopicHeadersContainer>
      <Date>
        <span>{moment().format("dddd | ")}</span>
        {moment().format("MMM DD, YYYY")}
      </Date>
      <TopicContainer>
        {width < 1024 ? (
          <>
            <ResponsiveMenu />
            {responsiveMenu ? <ResponsiveMenuModal /> : null}
          </> //
        ) : (
          <>
            {topics.map((topic) => {
              return (
                <StyledLink key={topic} to={`/${topic.toLowerCase()}`}>
                  <span
                    onClick={() => {
                      dispatch(changeHeader(topic.toLowerCase()));
                      dispatch(clearError());
                    }}
                  >
                    {topic}
                  </span>
                </StyledLink>
              );
            })}
          </> //
        )}
      </TopicContainer>
    </StoryTopicHeadersContainer>
  );
};

export default StoryTopics;
