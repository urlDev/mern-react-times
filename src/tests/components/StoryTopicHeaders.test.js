import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import moment from "moment";

import * as ReactReduxHooks from "utils/react-redux-hooks";
import { mockStore } from "tests/store";

import StoryTopicHeaders from "components/story-topic-headers/StoryTopicHeaders";
import ResponsiveMenuModal from "components/responsive-menu-modal/ResponsiveMenuModal";
import ResponsiveMenu from "components/responsive-menu/ResponsiveMenu";
import { CHANGE_HEADER, CLEAR_ERROR } from "redux/actions/news";

describe("Testing StoryTopicHeaders component", () => {
  let wrapper;
  let store;

  beforeEach(() => {
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation((state) => store.getState());

    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);

    // Mocking moment().format()
    Date.now = jest.fn(() => new Date("2021-01-16T00:00:00.000Z"));
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.resetAllMocks();
  });

  describe("Testing the component for when width is smaller than 1024", () => {
    beforeEach(() => {
      store = mockStore({
        width: 900,
        responsiveMenu: true,
      });

      wrapper = mount(
        <Router>
          <StoryTopicHeaders store={store} />
        </Router>
      );
    });
    test("Should render the component with ResponsiveMenu and modal", () => {
      expect(wrapper).toHaveLength(1);
      expect(
        wrapper.containsMatchingElement(<ResponsiveMenuModal />)
      ).toBeTruthy();
      expect(wrapper.containsMatchingElement(<ResponsiveMenu />)).toBeTruthy();
    });
    test("Should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
    test("Should not show modal if its false", () => {
      store = mockStore({
        width: 900,
        responsiveMenu: false,
      });

      wrapper = mount(<StoryTopicHeaders store={store} />);

      expect(
        wrapper.containsMatchingElement(<ResponsiveMenuModal />)
      ).toBeFalsy();
    });
  });

  describe("Testing component for when width is bigger than 1024", () => {
    beforeEach(() => {
      store = mockStore({
        width: 1100,
        responsiveMenu: true,
      });

      wrapper = mount(
        <Router>
          <StoryTopicHeaders store={store} />
        </Router>
      );
    });

    test("Should render the component with topics", () => {
      expect(wrapper).toHaveLength(1);
      // Having home in wrapper means topics are rendered successfully
      expect(wrapper.find("Home")).toBeTruthy();
    });
    test("Should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
    test("Should dispatch actions when clicked on a topic", () => {
      // I defined the actions in a span inside link
      // thats why I am clicking on one of them
      wrapper.find("span").at(1).simulate("click");

      const actions = store.getActions();

      expect(actions).toEqual([
        { type: CHANGE_HEADER, payload: "home" },
        { type: CLEAR_ERROR },
      ]);
    });
  });
});
