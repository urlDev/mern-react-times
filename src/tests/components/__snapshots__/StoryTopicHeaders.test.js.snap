// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StoryTopicHeaders component Testing component for when width is bigger than 1024 Should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <StoryTopicHeaders
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <styled.div>
        <div
          className="sc-gsTCUz eDDnMw"
        >
          <styled.p>
            <p
              className="sc-hKgILt iGvpmN"
            >
              <span>
                Tuesday | 
              </span>
              Jan 12, 2021
            </p>
          </styled.p>
          <styled.div>
            <div
              className="sc-dlfnbm kddwlQ"
            >
              <Styled(Link)
                key="Home"
                to="/home"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/home"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/home"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/home"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Home
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Arts"
                to="/arts"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/arts"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/arts"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/arts"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Arts
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Business"
                to="/business"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/business"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/business"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/business"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Business
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Health"
                to="/health"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/health"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/health"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/health"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Health
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Opinion"
                to="/opinion"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/opinion"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/opinion"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/opinion"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Opinion
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Politics"
                to="/politics"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/politics"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/politics"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/politics"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Politics
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Science"
                to="/science"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/science"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/science"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/science"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Science
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Sports"
                to="/sports"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/sports"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/sports"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/sports"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Sports
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Technology"
                to="/technology"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/technology"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/technology"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/technology"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Technology
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
              <Styled(Link)
                key="Travel"
                to="/travel"
              >
                <Link
                  className="sc-eCssSg eJegWd"
                  to="/travel"
                >
                  <LinkAnchor
                    className="sc-eCssSg eJegWd"
                    href="/travel"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-eCssSg eJegWd"
                      href="/travel"
                      onClick={[Function]}
                    >
                      <span
                        onClick={[Function]}
                      >
                        Travel
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </StoryTopicHeaders>
  </Router>
</BrowserRouter>
`;

exports[`Testing StoryTopicHeaders component Testing the component for when width is smaller than 1024 Should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <StoryTopicHeaders
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <styled.div>
        <div
          className="sc-gsTCUz eDDnMw"
        >
          <styled.p>
            <p
              className="sc-hKgILt iGvpmN"
            >
              <span>
                Tuesday | 
              </span>
              Jan 12, 2021
            </p>
          </styled.p>
          <styled.div>
            <div
              className="sc-dlfnbm kddwlQ"
            >
              <ResponsiveMenu>
                <styled.div
                  alt="hamburger menu"
                  onClick={[Function]}
                >
                  <div
                    alt="hamburger menu"
                    className="sc-bdfBwQ bUEalG"
                    onClick={[Function]}
                  >
                    <img
                      alt="hamburger icon"
                      src="hamburgerClose.svg"
                    />
                  </div>
                </styled.div>
              </ResponsiveMenu>
              <ResponsiveMenuModal>
                <styled.div
                  onClick={[Function]}
                >
                  <div
                    className="sc-jSgupP gylrLH"
                    onClick={[Function]}
                  >
                    <styled.div>
                      <div
                        className="sc-gKsewC cPEtrP"
                      >
                        <Styled(Link)
                          key="Home"
                          to="/home"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/home"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/home"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/home"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Home
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Arts"
                          to="/arts"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/arts"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/arts"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/arts"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Arts
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Business"
                          to="/business"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/business"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/business"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/business"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Business
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Health"
                          to="/health"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/health"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/health"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/health"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Health
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Opinion"
                          to="/opinion"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/opinion"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/opinion"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/opinion"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Opinion
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Politics"
                          to="/politics"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/politics"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/politics"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/politics"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Politics
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Science"
                          to="/science"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/science"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/science"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/science"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Science
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Sports"
                          to="/sports"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/sports"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/sports"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/sports"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Sports
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Technology"
                          to="/technology"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/technology"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/technology"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/technology"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Technology
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                        <Styled(Link)
                          key="Travel"
                          to="/travel"
                        >
                          <Link
                            className="sc-eCssSg eJegWd"
                            to="/travel"
                          >
                            <LinkAnchor
                              className="sc-eCssSg eJegWd"
                              href="/travel"
                              navigate={[Function]}
                            >
                              <a
                                className="sc-eCssSg eJegWd"
                                href="/travel"
                                onClick={[Function]}
                              >
                                <span
                                  onClick={[Function]}
                                >
                                  Travel
                                </span>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </Styled(Link)>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </ResponsiveMenuModal>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </StoryTopicHeaders>
  </Router>
</BrowserRouter>
`;
